base/iterators.jl

```diff
@@ -1055,7 +1059,7 @@ _prod_axes1(a, A) =
     throw(ArgumentError("Cannot compute indices for object of type $(typeof(a))"))
 
 ndims(p::ProductIterator) = length(axes(p))
-length(P::ProductIterator) = prod(size(P))
+length(P::ProductIterator) = reduce(checked_mul, size(P); init=1)
```

Comment:
Q: should this be a method of `checked_length` instead?
```suggestion
length(P::ProductIterator) = prod(size(P))
checked_length(P::ProductIterator) = reduce(checked_mul, size(P); init=1)
```

Reply:
See the previous discussion here: https://github.com/JuliaLang/julia/pull/43429#discussion_r771763047

I believe that `checked_mul` should be the default. My motivation was a lengthy debugging process when this code blew up:
```julia
if length(collection) < 100
	foreach(expensive_calculation, collection)
end
```

This seems like a bug in Julia, rather than a user error.